library/src/main/java/com/bumptech/glide/RequestBuilder.java

```diff
@@ -903,125 +931,156 @@ private Request buildRequestRecursive(
-          obtainRequest(
-              target,
-              targetListener,
-              thumbnailOptions,
-              coordinator,
-              transitionOptions,
-              getThumbnailPriority(priority),
-              overrideWidth,
-              overrideHeight);
-
-      coordinator.setRequests(fullRequest, thumbnailRequest);
-      return coordinator;
+      return buildThumbnailRequestForMultiplier(thumbSizeMultiplier,
+          target, targetListener, parentCoordinator, requestOptions,
+          transitionOptions, priority, overrideWidth, overrideHeight);
     } else {
       // Base case: no thumbnail.
       return obtainRequest(
-          target,
-          targetListener,
-          requestOptions,
-          parentCoordinator,
-          transitionOptions,
-          priority,
-          overrideWidth,
-          overrideHeight);
+          target, targetListener, parentCoordinator, requestOptions,
+          transitionOptions, priority, overrideWidth, overrideHeight);
     }
   }
 
-  private Request obtainRequest(
+  @NonNull
+  private Request buildThumbnailRequestRecursive(
+      RequestBuilder<TranscodeType> thumbnailBuilder,
       Target<TranscodeType> target,
       RequestListener<TranscodeType> targetListener,
+      @Nullable RequestCoordinator parentCoordinator,
+      RequestOptions parentRequestOptions,
+      TransitionOptions<?, ? super TranscodeType> transitionOptions,
+      Priority priority,
+      int overrideWidth,
+      int overrideHeight) {
+    RequestOptions thumbRequestOptions = thumbnailBuilder.requestOptions;
+    TransitionOptions<?, ? super TranscodeType> thumbTransitionOptions =
+        thumbnailBuilder.transitionOptions;
+
+    // Apply our transition by default to thumbnail requests but avoid overriding custom options
+    // that may have been applied on the thumbnail request explicitly.
+    if (thumbnailBuilder.isDefaultTransitionOptionsSet) {
+      thumbTransitionOptions = transitionOptions;
+    }
+
+    Priority thumbPriority = thumbRequestOptions.isPrioritySet()
+        ? thumbRequestOptions.getPriority() : getThumbnailPriority(priority);
+
+    int thumbOverrideWidth = thumbRequestOptions.getOverrideWidth();
+    int thumbOverrideHeight = thumbRequestOptions.getOverrideHeight();
+    if (Util.isValidDimensions(overrideWidth, overrideHeight)
+        && !thumbRequestOptions.isValidOverride()) {
+      thumbOverrideWidth = parentRequestOptions.getOverrideWidth();
+      thumbOverrideHeight = parentRequestOptions.getOverrideHeight();
+    }
+
+    ThumbnailRequestCoordinator coordinator = new ThumbnailRequestCoordinator(parentCoordinator);
+    Request fullRequest =
+        obtainRequest(
+            target,
+            targetListener,
+            coordinator,
+            parentRequestOptions,
+            transitionOptions,
+            priority,
+            overrideWidth,
+            overrideHeight
+        );
+    this.isThumbnailBuilt = true;
```

Comment:
It's easier to understand I think if this parameter is checked in the same method where it's used? Can you move the check in https://github.com/bumptech/glide/pull/2777/files#diff-a38a476f7aa7c143d4cde2716bb2b59dR955 to this method?